### PR TITLE
[form-builder] Respect editModal schema option on Block Editor

### PR DESCRIPTION
**Pull request by kristofferj**
Respect editModal option in schema options (like on array), and choose proper edit modal when not set, based on number of fields in the type.

### CODE REVIEW

File: packages/@sanity/form-builder/src/inputs/BlockEditor-slate/FormBuilderBlock.js [hunk 1]
```diff
@@ -253,21 +255,77 @@ export default class FormBuilderBlock extends React.Component {
     const value = this.getValue()
     const memberType = this.getMemberTypeOf(value)
 
+    const fieldsQty = (get(memberType, 'fields') || []).length
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/inputs/BlockEditor-slate/FormBuilderBlock.js]: using lodash.get is overkill here. I'd prefer `const fieldsQty = (memberType && memberType.fields || []).length`

**Comment by bjoerge** [comment]
A minor nitpick, otherwise LGTM.

There's a bit of duplication on the part where we resolve which mode to use, could this be extracted to a shared component? I imagine we'd like to use the same heuristics in the standard array component as well?

File: packages/@sanity/form-builder/src/inputs/BlockEditor-slate/FormBuilderBlock.js (hunk 1 above)
Comment (kristofferj) [reply packages/@sanity/form-builder/src/inputs/BlockEditor-slate/FormBuilderBlock.js]: 👍

**Comment by kristofferj** [comment]
A shared component is a good idea. Exceeds the scope of this branch, but lets make it happen in another branch.

**Pull request merged**